pkg/host/sriov.go

```diff
@@ -0,0 +1,443 @@
+package host
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jaypipes/ghw"
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	dputils "github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/global/consts"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/global/vars"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+	mlx "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/vendors/mellanox"
+)
+
+func (h *HostManager) SetSriovNumVfs(pciAddr string, numVfs int) error {
+	log.Log.V(2).Info("SetSriovNumVfs(): set NumVfs", "device", pciAddr, "numVfs", numVfs)
+	numVfsFilePath := filepath.Join(vars.FilesystemRoot, consts.SysBusPciDevices, pciAddr, consts.NumVfsFile)
+	bs := []byte(strconv.Itoa(numVfs))
+	err := os.WriteFile(numVfsFilePath, []byte("0"), os.ModeAppend)
+	if err != nil {
+		log.Log.Error(err, "SetSriovNumVfs(): fail to reset NumVfs file", "path", numVfsFilePath)
+		return err
+	}
+	err = os.WriteFile(numVfsFilePath, bs, os.ModeAppend)
+	if err != nil {
+		log.Log.Error(err, "SetSriovNumVfs(): fail to set NumVfs file", "path", numVfsFilePath)
+		return err
+	}
+	return nil
+}
+
+func (h *HostManager) ResetSriovDevice(ifaceStatus sriovnetworkv1.InterfaceExt) error {
+	log.Log.V(2).Info("ResetSriovDevice(): reset SRIOV device", "address", ifaceStatus.PciAddress)
+	if err := h.SetSriovNumVfs(ifaceStatus.PciAddress, 0); err != nil {
+		return err
+	}
+	if ifaceStatus.LinkType == consts.LinkTypeETH {
+		var mtu int
+		is := sriovnetworkv1.InitialState.GetInterfaceStateByPciAddress(ifaceStatus.PciAddress)
+		if is != nil {
+			mtu = is.Mtu
+		} else {
+			mtu = 1500
+		}
+		log.Log.V(2).Info("ResetSriovDevice(): reset mtu", "value", mtu)
+		if err := h.SetNetdevMTU(ifaceStatus.PciAddress, mtu); err != nil {
+			return err
+		}
+	} else if ifaceStatus.LinkType == consts.LinkTypeIB {
```

Comment:
for IB we dont rely on initial state ?

Reply:
not today https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/pkg/utils/utils.go#L697